src/instance.rs

```diff
@@ -25,18 +25,130 @@ pub trait PyObjectWithGIL: Sized {
 #[doc(hidden)]
 pub trait PyNativeType: PyObjectWithGIL {}
 
+/// A reference of type `T` which is in Python heap.
```

Comment:
The docs should mention that `T` is part of a python object if it's a PyRef[Mut].

Reply:
:+1: